system/jlib/jstring.cpp

```diff
@@ -2482,3 +2501,49 @@ const char * nullText(const char * text)
     if (text) return text;
     return "(null)";
 }
+
+StringBuffer& StringBuffer::operator=(StringBuffer&& value)
+{
+    swapWith(value);
+    return *this;
+}
+
+StringBuffer operator+ (const StringBuffer& lhs, const StringBuffer& rhs)
```

Comment:
The use of the operator+ is a reasonably efficient and intuitive way to do string concatenation.  It is inherently less error prone than variadic functions VStringBuffer, setf, appendf.  Also concatenated StringBuffers can be created directly as parameters to functions.  Something like this would now be possible: 

```
void foo(const char *);  
….
StringBuffer mystr,yourstr;
…..
foo( mystr + yourstr + propertyTree->queryProp(“@name”) );
foo( StringBuffer("MyString") + mystr + yourstr + "last" );
…
```

This simplifies the typical approach of creating a named temporary StringBufffer object and appending to it.  It also eliminates the alternate approach of creating a VStringBuffer either as an r or l value.  This syntax is also likely to lead to fewer lines of code to maintain.